### ISSUE TITLE
[New Rule] Enumeration of 4 or more variables

### ISSUE THREAD

**Issue by MartinThoma** [MartinThoma/flake8-simplify#133]
## Explanation

A code smell I see sometimes is enumeration of variables where an iterable should be used, e.g. a list.

Using a list has several advantages:

* Extendability: If foo1, foo2, foo3, foo4 exist, maybe there will also be foo5 in future?
* You can iterate: If you want to do an action on all of the foo-variables, it's easier
* Maintainability: Renaming foo is easier than foo1-foo4; passing foo as a parameter is easier than passing foo1-4

## Example

```python
# Bad
foo1 = "bar"
foo2 = "baz"
foo3 = "42"
foo4 = "1337"

# Good
foo = ["bar", "baz", "42", "1337"]
```

## False-Positives

Just looking for a number in the end would have a quite high chance of false-positivies, e.g.:

* Up to 3 might be acceptable, e.g. for coordinates `x1` , `x2`, `x3` is not too uncommon.
* Some names could have 4 at the end, e.g. `ieee754`

Hence I would look for variables where the direct siblings are 1-4 (or more)
